### PR TITLE
CHANGELOG for ACR OAuth2 auth support / refactor and clean up code

### PR DESCRIPTION
**Pull request by chanseokoh**
After #1511.

### CODE REVIEW

File: jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java [hunk 1]
```diff
@@ -50,23 +59,50 @@ public void testFromAuthenticationMethod_bearer()
...
-    }
+  public void testAuthRequestParameters_basicAuth() {
+    Assert.assertEquals(
+        "service=someservice&scope=repository:someimage:scope",
+        registryAuthenticator.getAuthRequestParameters("scope"));
+  }
+
+  @Test
+  public void testAuthRequestParameters_oauth2() {
+    registryAuthenticator.setCredential(Credential.basic("<token>", "oauth2_access_token"));
```
Comment (andxu) [inline jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java]: shall we rename the method : Credential.basic ?

Comment (chanseokoh) [reply jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java]: I just tried this, and it touches too many files. I filed an issue https://github.com/GoogleContainerTools/jib/issues/1526, and I'll create a PR for that soon.

**Comment by andxu** [comment]
@chanseokoh with this oauth2 change, now the azure acr-docker-credential-helper is supported, could you please also update the document part of support azure acr helper? the steps of using acr helper:

see doc at https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-azure-cli
1. create acr and login,  eg: name :jibtest
```
az group create --name myResourceGroup --location eastus
az acr create --resource-group myResourceGroup --name jibtest--sku Basic
az acr login -n jibtest
```
2. install acr helper

For Windows, run the [powershell installation script](https://aka.ms/acr/installaad/win) in administrator mode:

```iex ([System.Text.Encoding]::UTF8.GetString((Invoke-WebRequest -Uri https://aka.ms/acr/installaad/win).Content))```

For Linux and macOS, run the [bash installation script](https://aka.ms/acr/installaad/bash):

```curl -L https://aka.ms/acr/installaad/bash | /bin/bash```

3.  Configure jib 


```
  to {
    image = 'jibtest.azurecr.io/testimage'
    credHelper = 'acr-windows'
    tags = [ 'latest']
  }
```
```
<to>
  <image>jibtest.azurecr.io/testimage:latest</image>
  <credHelper>acr-windows</credHelper>
</to>
```

**Comment by chanseokoh** [comment]
@andxu thanks for the doc suggestion! Filed an issue #1530, and we will get to it soon.

@GoogleContainerTools/java-tools I think we can cut the next release after merging this.

**Pull request merged**